oxidation/libparsec/crates/client_types/src/native/local_device_file.rs

```diff
@@ -0,0 +1,214 @@
+                    device.human_handle,
+                    device.device_label,
+                    device.slug,
+                ),
+                DeviceFile::Smartcard(device) => (
+                    DeviceFileType::Smartcard,
+                    device.organization_id,
+                    device.device_id,
+                    device.human_handle,
+                    device.device_label,
+                    device.slug,
+                ),
+            };
+
+        Ok(Self {
+            key_file_path: key_file_path.into(),
+            organization_id,
+            device_id,
+            human_handle,
+            device_label,
+            slug,
+            ty,
+        })
+    }
+}
+
+/// Return the default keyfile path for a given device.
+///
+/// Note that the filename does not carry any intrinsic meaning.
+/// Here, we simply use the slughash to avoid name collision.
+pub fn get_default_key_file(config_dir: &Path, device: &LocalDevice) -> PathBuf {
+    let devices_dir = config_dir.join("devices");
+    let _ = std::fs::create_dir_all(&devices_dir);
+    devices_dir.join(device.slughash() + ".keys")
+}
+
+pub async fn save_recovery_device(
+    key_file: &Path,
+    device: LocalDevice,
+    force: bool,
+) -> LocalDeviceResult<String> {
+    if File::open(key_file).is_ok() && !force {
+        return Err(LocalDeviceError::AlreadyExists(key_file.to_path_buf()));
+    }
+
+    let (passphrase, key) = SecretKey::generate_recovery_passphrase();
+
+    let ciphertext = key.encrypt(&device.dump());
+
+    let key_file_content = DeviceFile::Recovery(DeviceFileRecovery {
+        ciphertext,
+        organization_id: device.organization_id().clone(),
+        slug: device.slug(),
+        human_handle: device.human_handle,
+        device_label: device.device_label,
+        device_id: device.device_id,
+    });
+
+    key_file_content.save(key_file)?;
+
+    Ok(passphrase)
+}
+
+/// TODO: need test (backend_cmds required)
+pub async fn load_recovery_device(
+    key_file: &Path,
+    passphrase: &str,
+) -> LocalDeviceResult<LocalDevice> {
+    let ciphertext =
+        std::fs::read(key_file).map_err(|_| LocalDeviceError::Access(key_file.to_path_buf()))?;
+    let data = rmp_serde::from_slice::<DeviceFile>(&ciphertext)
+        .map_err(|_| LocalDeviceError::Deserialization(key_file.to_path_buf()))?;
+
+    let device = match data {
+        DeviceFile::Recovery(device) => device,
+        _ => return Err(LocalDeviceError::Validation(DeviceFileType::Recovery)),
+    };
+
+    let key =
+        SecretKey::from_recovery_passphrase(passphrase).map_err(LocalDeviceError::CryptoError)?;
+    let plaintext = key
+        .decrypt(&device.ciphertext)
+        .map_err(LocalDeviceError::CryptoError)?;
+    LocalDevice::load(&plaintext)
+        .map_err(|_| LocalDeviceError::Deserialization(key_file.to_path_buf()))
+}
+
+/// TODO: Remove me once wasm doesn't need to mock
+pub fn test_gen_default_devices() {}
```

Comment:
I guess we should add a `panic!` in this function. Otherwise the behavior between wasm and native is not the same in a not obvious way from the frontend developer (he is going to develop with wasm, then see the test he wrote doesn't work in native... at this point we need to provide him an obvious feedback)

Reply:
Why do we even include this function, it's only used by `wasm` so we can remove it ?